packages/jbrowse-plugin-apollo/src/components/AddAssembly.tsx

```diff
@@ -0,0 +1,115 @@
+import { AbstractSessionModel, AppRootModel } from '@jbrowse/core/util'
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  TextField,
+} from '@material-ui/core'
+import { getRoot } from 'mobx-state-tree'
+import React, { useState } from 'react'
+
+import { ApolloInternetAccountModel } from '../ApolloInternetAccount/model'
+import { ViewModel } from './stateModel'
+
+interface AddAssemblyProps {
+  session: AbstractSessionModel
+  handleClose(): void
+}
+
+export function AddAssembly({ session, handleClose }: AddAssemblyProps) {
+  // export function AddAssembly({ session, handleClose }: AddAssemblyProps) {
+  const { internetAccounts } = getRoot(session) as AppRootModel
+  const apolloInternetAccount = internetAccounts.find(
+    (ia) => ia.type === 'ApolloInternetAccount',
+  ) as ApolloInternetAccountModel | undefined
+  if (!apolloInternetAccount) {
+    throw new Error('No Apollo internet account found')
+  }
+  const { baseURL, internetAccountId } = apolloInternetAccount
+  const [assemblyName, setAssemblyName] = useState('')
+//   const [setAssemblyDesc, setAssemblyDesc] = useState('')
+
+  async function onSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+    const sessionToken = sessionStorage.getItem('apolloInternetAccount-token')
+    if (sessionToken == null) {
+      alert('You must authenticate first!')
+      return
+    }
+    console.log(`Assembly name is "${assemblyName}"`)
+
+    const res = await fetch(new URL('/changes/submitChange', baseURL).href, {
```

Comment:
Instead of getting the token manually, you'll need to get the custom fetcher from the internet account. That would look something like this:

```js
const uri = new URL('/changes/submitChange', baseURL).href
const apolloFetch = apolloInternetAccount.getFetcher({
  locationType: 'UriLocation',
  uri,
})
const res = await apolloFetch(uri, { //...
```

That way you don't have to add the authorization  header, it'll get added for you automatically.